packages/codegen/src/scanEntityFiles.ts

```diff
@@ -13,13 +21,7 @@ export async function scanEntityFiles(config: Config, dbMeta: DbMetadata): Promi
       try {
         // load the entity.ts file (as a promise with fs/promises)
         const tsCode = (await readFile(join(config.entitiesDirectory, `${entity.name}.ts`))).toString();
-        const defaultableFields = [
-          ...entity.primitives,
-          ...entity.enums,
-          ...entity.pgEnums,
-          ...entity.manyToOnes,
-          ...entity.polymorphics,
-        ];
+        const defaultableFields = recursiveDefaultableFields(dbMeta, entity);
```

Comment:
@chr1sjf0x is the fix here for a field is defined in a base class / base table, but the `setDefault` call lives only in a specific subtype?

Reply:
Yes. If you have a field which is shared between the subtypes, so the field itself is defined on the base class, but the subtypes provide different `setDefault` values (and there is no default in the base), then this code used to fail to detect that it should consider the base field to have a default.

Per the note below, I suspect there are issues here, like if one subtype defines a default but another doesn't. Or maybe one subtype wants the field to be async derived while the other wants it editable or sync derived. These sort of cases would need a larger refactor than I was willing to undertake.